src/main.rs

```diff
@@ -226,6 +244,31 @@ struct Counter {
     step_by: f64,
 }
 
+#[derive(Debug)]
+struct Summary {
+    successes: u32,
+    failures: Vec<u32>
+}
+
+impl Summary {
+    fn print(self) {
+        let total = self.successes + self.failures.len() as u32;
+
+        let errors = if self.failures.is_empty() {
+            String::from("0")
+        } else {
+            self.failures.into_iter()
```

Comment:
It'd be better if this could include a summary in addition to the codes themselves, so `2 (-1, -1)` rather than just `-1, -1`.

Reply:
Done 😄